testing/jormungandr-testing-utils/src/testing/network_builder/wallet/template/mod.rs

```diff
@@ -13,30 +14,49 @@ pub struct WalletTemplate {
     value: Value,
     wallet_type: WalletType,
     delegate: Option<NodeAlias>,
+    discrimination: Discrimination,
 }
 
 impl WalletTemplate {
-    pub fn new_account<S: Into<WalletAlias>>(alias: S, value: Value) -> Self {
-        Self::new(alias, value, WalletType::Account)
+    pub fn new_account<S: Into<WalletAlias>>(
+        alias: S,
+        value: Value,
+        discrimination: Discrimination,
+    ) -> Self {
+        Self::new(alias, value, WalletType::Account, discrimination)
     }
-    pub fn new_utxo<S: Into<WalletAlias>>(alias: S, value: Value) -> Self {
-        Self::new(alias, value, WalletType::UTxO)
+    pub fn new_utxo<S: Into<WalletAlias>>(
+        alias: S,
+        value: Value,
+        discrimination: Discrimination,
+    ) -> Self {
+        Self::new(alias, value, WalletType::UTxO, discrimination)
     }
 
     #[inline]
-    fn new<S: Into<WalletAlias>>(alias: S, value: Value, wallet_type: WalletType) -> Self {
+    fn new<S: Into<WalletAlias>>(
+        alias: S,
+        value: Value,
+        wallet_type: WalletType,
+        discrimination: Discrimination,
+    ) -> Self {
```

Comment:
Looks like another case for a builder API, especially if most tests would like the default discrimination (`Test`)?